### PR TITLE
[libc] remove header guard for assert.h

### PR DESCRIPTION
**Pull request by nickdesaulniers**
It's meant to be included multiple times! Maybe use a NOLINT rule to suppress
clang-tidy's llvm-header-guard lint warning.

### CODE REVIEW

File: libc/src/assert/assert.h [hunk 1]
```diff
@@ -1,3 +1,4 @@
+// NOLINT(llvm-header-guard)
```
Comment (nickdesaulniers) [inline libc/src/assert/assert.h:1]: Note to reviewers: It seems like this MUST be the first line FWICT.

Comment (jhuber6) [reply libc/src/assert/assert.h:1]: That's silly, considering that every single file in LLVM starts with a header comment. I'd open an issue personally.

Comment (michaelrj-google) [reply libc/src/assert/assert.h:1]: agreed, but there's not much to do for the moment. Perhaps add a todo to move this when possible?

**Review by michaelrj-google — APPROVED** [review]

File: libc/src/assert/assert.h (hunk 1 above)
Comment (nickdesaulniers) [reply libc/src/assert/assert.h]: Done in https://github.com/llvm/llvm-project/pull/83334/commits/8ef16ad9a4cde241f5c482609ff6549e2b2f7df4 (https://github.com/llvm/llvm-project/issues/83339)

**Pull request merged**